### ISSUE TITLE
torch.solver.KolSol() doesn't seem to handle autograd

### ISSUE THREAD

**Issue by danielkelshaw** [MagriLab/KolSol#48]
When using `kolsol.torch.solver.KolSol()` in an environment where backward-differentiation needs to be performed through the solver, the gradients come out to `nan` in almost every case. There are perhaps two things to look at:

**One:**

https://github.com/MagriLab/KolSol/blob/f6c6e1ab07c533166e90f304535d92bd117c71d9/kolsol/torch/solver.py#L83

This line of code may not play nicely with `autograd` -- may need to look at a tensor-operation based alternative.

**Two:**

May need to explicitly state that the tensors used within the solver have: `requires_grad=True`

**Comment by danielkelshaw** [comment]
This problem seems to arrive because of a division-by-zero error. As the element being divided by zero is then replaced by `0.0` afterwards anyway, we can just change the divisor to divide-by-one in that instance to avoid this problem.
